File: src/main/java/com/rpb/reservation/reservation/application/ReservationShareInfo.java
package com.rpb.reservation.reservation.application;

import java.util.UUID;

public record ReservationShareInfo(
    UUID reservationId,
    String reservationNo,
    String channel,
    String shareText,
    String customerMaskedPhone,
    boolean customerPhoneAvailable,
    String senderLabel,
    boolean canOpenWhatsAppLink,
    String whatsappLink,
    boolean canOpenWechatLink,
    String wechatLink,
    String wechatShareText,
    String shareToken,
    String sharePath,
    String shareTitle,
    String shareSummary
) {
}

File: src/types/reservationShareInfo.ts
export interface ReservationShareInfo {
  reservationId: string
  reservationNo: string
  channel: 'manual_copy' | (string & {})
  shareText: string
  customerMaskedPhone: string
  customerPhoneAvailable: boolean
  senderLabel: string
  canOpenWhatsAppLink: boolean
  whatsappLink?: string | null
  canOpenWechatLink: boolean
  wechatLink?: string | null
  wechatShareText?: string | null
  shareToken: string
  sharePath: string
  shareTitle: string
  shareSummary: string
}

export interface ReservationShareInfoResponse {
  success: true
  shareInfo: ReservationShareInfo
}

export type ReservationShareIntentChannel = 'whatsapp' | 'wechat' | 'system_share' | 'copy_link'

export interface ReservationShareIntentResponse {
  success: true
  channel: ReservationShareIntentChannel
}

export interface ReservationShareInfoApiErrorResponse {
  success: false
  error: {
    code: string
    messageKey: string
    details: Record<string, unknown>
  }
}
